chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -2268,7 +2240,11 @@ impl PeerManagerActor {
                 PeerManagerMessageResponse::OutboundTcpConnect(())
             }
             PeerManagerMessageRequest::InboundTcpConnect(msg) => {
-                self.handle_msg_inbound_tcp_connect(msg, ctx);
+                if self.peer_counter.load(Ordering::SeqCst)
+                    < self.config.max_num_peers as usize + LIMIT_PENDING_PEERS
+                {
+                    self.handle_msg_inbound_tcp_connect(msg, ctx);
+                }
```

Comment:
so here we'd silently drop the connection if the limit is exceeded? might make debugging a lot harder. 

We should add some log here (and make sure that it is not printed too often)

Reply:
Yes, that's how the current logic works. We can discuss changing it in another PR.